### DISCUSSION TITLE
GitHub-hosted larger runners for Actions are generally available

### DISCUSSION THREAD

**Discussion by tuves** [community/community#58739] [Enterprise] [answered]
### Select Topic Area

Product Feedback

### Body

We are announcing that larger hosted runners for GitHub Actions are generally available for paid Team and Enterprise Cloud plans! This feature has been in public beta since [September of 2022](https://github.blog/2022-09-01-github-actions-introducing-the-new-larger-github-hosted-runners-beta/) where customers have been using it in production to run their CI/CD jobs faster and with more flexibility.

![image](https://github.com/community/community/assets/49915749/951f12b8-b956-4884-97f8-3e50cde19162)
The new larger runners provide new capabilities:

**Larger Linux and Windows machines:** 
This allows development teams to use machine sizes up to 64 vCPUs with 256 GB of RAM, and 2 TB of SSD storage to support their on-demand CI/CD jobs and other workflows. Larger runners are charged by the job minute for both private and public repositories and do not consume included minutes.

**Static IP address to enable secure access to your resources:** 
Enterprise Cloud customers can now choose whether a static IP address range will be assigned to their larger runner instances. This provides a fixed IP address range that you can add to your allow list for access to internal systems. You can also use this in conjunction with GitHub’s IP allow list to enable hosted actions runners and IP allow listing at the same time.

**Administrative control over access to larger runners and concurrency:** 
Your administrators can decide who has access to larger machine sizes and at what concurrency, providing guard rails on spending.

You can learn about the larger runner per job minute pricing by checking the [current pricing documentation](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) and learn more about this feature by digging into the [documentation](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners).

**Comment by Vichugo1995** [comment]
Hola cómo puedo recuperar dinero de una app estafa

**Comment by amianicaroline** [comment]
hi i am carol from Nairobi Kenya.and i would like to ask a quiz all GitHub

**Comment by amianicaroline** [comment]
plz i Need help

**Comment by eveenendaal** [comment]
Any plans to make this available for paid individual plans?

**Reply by davidaucoin7377** [reply]
I would love to make plans . But I need guidance in the right direction on what I need to do next on github. Please I need a AI guidance? My email...Davidaucoin7377@gmail.com  help me please some body?

**Reply by q-o-ap** [reply]
the individual  there no fax when use the plan

**Reply by tanmayeekamath** [reply]
Hey @eveenendaal, we do have plans to make the larger skus available for Individual plans, however I don't have a concrete timeline to share just yet. Noting your feedback as a +1 on our backlog, we'll update [our public roadmap](https://github.com/orgs/github/projects/4247/views/1?filterQuery=-status%3A%22Q4+2022+%E2%80%93+Oct-Dec%22++label%3Arunners) as soon as we have the timeline. 

Also curious to hear what type of scenarios this would enable for you, and which SKUs/OSes/functionalities would be critical for the same, can you share more on that?:)

**Reply by eveenendaal** [reply]
I'd like the larger instances to speed up my builds. Mostly for my builds that include Cypress tests. Cypress includes starting up a whole headless chromium instance, so the larger instance would speed that up quite a bit.

**Reply by intUnderflow** [reply]
Would also love to be able to use larger runners as an individual @tanmayeekamath, I'm trying to use GitHub actions to run a backup task that uses pg_dumpall. Right now the only solution is to use a self-hosted runner which means reserving a massive amount of disk space on that runner at all times, when its only needed once a day.

**Reply by m95e** [reply]
Write the new code model get sources of metal to build at home

**Comment by Puffylow** [comment]
Im a single mom 2 underage girls. I was being watched by someone using this coding program.  Im a private resident. He took pictures.  How can I find him and was this created for watching private homes? Please help. They will not stop. It's everyday. I've changed phones, sim cards, and networks. My daughter's can not come to my home because it is not safe.  I saw a username Astronaut. Please please help!!!By the way.  I'm not @Puffylow. They have been on my phone for months.  No one would help or stop this person.  When I got on to this site this was the name of the account I assume I created when signing in.  I don't know, but someone here must be able to help.  I have no where else to run. I have never heard of this before and definitely never consented to any of this. Please help

**Comment by Puffylow** [comment]
![image000001(2)](https://github.com/community/community/assets/130266656/0b0fc7e4-18cf-40b1-8923-7c25ebe543fc)
![image000002(2)](https://github.com/community/community/assets/130266656/2a36fa20-dce2-4533-92f9-c5c86dddc4a4)
![image000003(2)](https://github.com/community/community/assets/130266656/179751d4-64f3-40db-801a-9f1fffee43c8)
![image000004(2)](https://github.com/community/community/assets/130266656/cc96d813-77cb-4b0f-9035-e89fcd66bdf4)
![image000005(1)](https://github.com/community/community/assets/130266656/b1151e62-2acc-43ed-b882-50b940c8cbee)
![image000006(3)](https://github.com/community/community/assets/130266656/e8938c7d-40ba-473c-9f34-fed9a8c6eb09)
![image000007(1)](https://github.com/community/community/assets/130266656/684d0caa-eac9-40b6-b806-1fc56801d52a)
![image000008(3)](https://github.com/community/community/assets/130266656/12482a9d-bdcb-4a4c-a10b-4aea7273f97a)
![image000001(3)](https://github.com/community/community/assets/130266656/fc8def6a-d9b3-42ad-b264-062d7d16d82e)
![image000002(3)](https://github.com/community/community/assets/130266656/f28db4a9-3083-4fcd-82a8-1897dffade22)
![image000007(2)](https://github.com/community/community/assets/130266656/107d1f66-c14c-493d-848f-65089d50d1dd)
![image000009(4)](https://github.com/community/community/assets/130266656/36319605-a83b-40d0-bf5e-383a7769aad9)
![image000000(6)](https://github.com/community/community/assets/130266656/3fa27ead-c29b-4cff-80d0-ddc8aadde1e2)
![image000002(4)](https://github.com/community/community/assets/130266656/bc6b26ad-d822-43d8-a3f9-e853f4d3c735)
![image000001(4)](https://github.com/community/community/assets/130266656/758e3a18-de88-4c16-8a05-69ffcf10a6d5)
![image000003(3)](https://github.com/community/community/assets/130266656/beabe7a7-51da-4159-be3e-bf2972e076ab)
![image000004(3)](https://github.com/community/community/assets/130266656/534eb7cc-d5e7-415c-a38c-cf2a7bac0ca8)
![image000006(2)](https://github.com/community/community/assets/130266656/d53f46f2-e053-4401-afc1-42fbef4197f1)
![image000005(2)](https://github.com/community/community/assets/130266656/99e82251-930e-4535-9745-efa5e70453d2)
![image000007](https://github.com/community/community/assets/130266656/25432b09-f5b2-4389-804c-778c037ab97d)
![image000008
![image000003](https://github.com/community/community/assets/130266656/2cf34bde-6c76-4740-9d0d-6ab26ba21d59)
![image000004(1)](https://github.com/community/community/assets/130266656/42318d95-e00b-4c3a-8475-3dcd427fb069)
(2)](https://github.com/community/community/assets/130266656/41dd1f29-580e-46cc-b554-76042903f56c)
![image000000(4)](https://github.com/community/community/assets/130266656/67e0173a-20ec-4de3-aaf0-dba52120c019)
![image000009(2)](https://github.com/community/community/assets/130266656/4dc1f5bb-cb56-461d-86e2-cf96b5e96a5b)
![image000000(1)](https://github.com/community/community/assets/130266656/d62a2402-6526-442b-bb54-624cfea404a4)
![image000002(1)](https://github.com/community/community/assets/130266656/48137b5d-b3ff-47ce-b389-cb1a7a8f8199)

**Reply by Puffylow** [reply]
I have more but they are deleting me out of here. Please please help me.  I want to know who and why they are doing this.  I live alone and I am a divorced woman with 2 kids.  I'm not a company and I am a nobody.  They will turn off my data usage and wifi so at night I can't even call for help.  Please help 
Jessicanittel@yahoo.com 864-621-0574.  Please someone help me.  I have more pictures but they changed my passwords for my emails, Microsoft and Samsung.  Which I created Microsoft looking for them.  This has been a horrible experience.  I'm afraid.  I thought it was a criminal., why are they doing this to Mr. I've never heard of any of this.  I never gave anyone permission for any of this.  please please help

**Comment by nashetali837** [comment]
Any plans to make this available for paid individual plans?

**Comment by ADHULE** [comment]
hello the community am a beginner in programming, because am a student by this occasion I would like to ask you how to communicate the arduino card with the eclipse? please help me

**Comment by ADHULE** [comment]
hello the community am a beginner in programming, because am a student by this occasion I would like to ask you how to communicate the arduino card with the eclipse? please help me

**Comment by alexisLefebvre** [comment]
I don't know where to address this but, in the announcement https://github.blog/changelog/2023-09-06-dual-static-ip-ranges-for-github-hosted-larger-runners/ “recieve” is written instead of “receive”, 2 times.

**Reply by tuves** [reply]
Great catch, @alexislefebvre! The changelog has been updated :)

**Comment by XoL1507** [comment]
first i need to meet either 2,3 of you in person please. i'm hoping.

**Comment by fajterini** [comment]
Hi, yesterday there was announcement about [ Azure private networking for GitHub](https://github.blog/changelog/2023-11-01-github-hosted-runners-private-networking-with-azure-virtual-networks-public-beta/) for (large runners). 
I was trying to look also into documentation, and we have some questions: 

1,) Supported regions include East US, East US 2, and West US 2.  - when will this be available in Europe ? This are supported Azure locations ?
2.) The GitHub hosted runners - where are they located (geo) - this becomes a topic in relation to data residency and last I checked, Github still has no control over this. We need that our data stay in EU. Can this be configured ?
3.) The solution implements essentially a service endpoint to the VNet ? - what about traffic visibility (TLS inspection, firewall control etc.)? Can you give better details how this is implemented in Azure ?  Also some architecture overview and diagram how data will be moving. I was checking this [ link ](https://docs.github.com/en/enterprise-cloud@latest/admin/configuration/configuring-private-networking-for-hosted-compute-products/configuring-private-networking-for-github-hosted-runners) but I haven't found this information. 
4.) If I understand correctly runners will use dedicated vnet subnet that will be used just by this resource ?

**Reply by tanmayeekamath** [reply, marked as answer]
Hey @fajterini thanks for reaching out. 

1,) Supported regions include East US, East US 2, and West US 2. - when will this be available in Europe ? This are supported Azure locations?
[Tanmayee] We don't have a timeline for Europe regions just yet due to Azure capacity restrictions in those regions. Can you please fill out this [form](https://resources.github.com/private-networking-for-github-hosted-runners-with-azure-virtual-networks/) to share with us specific EU regions that will be critical for you? Thanks!

2.) The GitHub hosted runners - where are they located (geo) - this becomes a topic in relation to data residency and last I checked, Github still has no control over this. We need that our data stay in EU. Can this be configured?
[Tanmayee] With the Azure VNET private networking functionality, data residency will be supported since the VM resource and the corresponding injected NIC resource need to be in the same region. For example, if you connect a North EU VNET to your GitHub Enterprise, then all your data for GitHub-hosted runners will remain within that region. 

For 3) Here's a link to an architecture diagram, does this help?
https://docs.github.com/en/enterprise-cloud@latest/admin/configuration/configuring-private-networking-for-hosted-compute-products/about-using-github-hosted-runners-in-your-azure-virtual-network#about-network-communication
In terms of firewall control or any other network policies, customers will have full control over it with the VM's NIC being deployed in your network. 

4.) If I understand correctly runners will use dedicated vnet subnet that will be used just by this resource ?
Correct, runner NICs will be deployed into a dedicated VNET-subnet that is owned and managed by the customer. 

Hope this helps, feel free to respond with any follow up questions:)

**Reply by fajterini** [reply]
Thanks, for good and fast response :) 
I think when this will be available in our region West EU we could use this new feature, but I would still prefer native support for Azure VMSS like Azure DevOps have.

Best regards,
M.

**Reply by Jorge-Rodriguez** [reply]
Really looking forward to West Europe support too.

Cheers!

**Reply by tungpe** [reply]
**500000**

**Comment by alexymantha** [comment]
Hi, any plans to make this configurable through the GitHub API?

**Reply by jndz2** [reply]
It would be great if this could be made configurable via the API.

**Reply by Laffs2k5** [reply]
+1 for support in the REST API

**Comment by jokerputihboy** [comment]
Jokerboy999
